### PR TITLE
ocamlPackages.mec: init at 0.1.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Broken out from #195344

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/mec/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "sha256-uIcGj/exSfuuzsv6C/bnJXpYRu3OY3dcKMW/7+qwi2U=";
+  };
+
+  propagatedBuildInputs = [
+    eqaf
+    bigarray-compat
+    hex
+    ff-sig
+    ff
+    alcotest
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/mec/default.nix:19]: This looks wrong but is actually correct

Comment (vbgl) [reply pkgs/development/ocaml-modules/mec/default.nix:21]: Is it really correct or is it an upstream bug (that you may report)?

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/mec/default.nix:21]: It's actually correct, the usage is here: https://gitlab.com/nomadic-labs/cryptography/ocaml-ec/-/blob/master/src/curve/ec_pbt.ml#L21

Comment (vbgl) [reply pkgs/development/ocaml-modules/mec/default.nix:21]: Suprising design.

File: pkgs/development/ocaml-modules/mec/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  checkInputs = [
+    alcotest
+    bisect_ppx
+  ];
+
+  meta = {
+    description = "Mec - Mini Elliptic Curve library";
+    license = lib.licenses.mit;
+    platforms = ocaml.meta.platforms;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/mec/default.nix:36]: The `platforms` attribute is set by `buildDunePackage`; yet `homepage` seems missing.

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/mec/default.nix]: Fixed

**Pull request merged**